kernel/arch/x86_64/boot/idt_dispatch.c

```
@@ -0,0 +1,25 @@
+
+int x86_idt_register_handler(int vector, void (*handler)(void)) {
+    if (vector < 0 || vector >= 256 || !handler)
+        return -1;
+    s_handlers[vector] = handler;
+    return 0;
+}
+
+/* Called from isr_common_stub with the vector number in %rdi (SysV AMD64 ABI).
+ * Interrupts are disabled by the CPU on entry to any interrupt gate. */
+void x86_idt_dispatch(uint64_t vector) {
+    if (vector < 256u && s_handlers[vector])
```

Clang (14.0.6)

[warning] integer literal has suffix 'u', which is not uppercase

(readability-uppercase-literal-suffix)